charms/argo-controller/tests/integration/test_charm.py

```diff
@@ -137,17 +137,22 @@ async def test_prometheus_grafana_integration(ops_test: OpsTest):
     prometheus_scrape_charm = "prometheus-scrape-config-k8s"
     scrape_config = {"scrape_interval": "30s"}
 
-    await ops_test.model.deploy(prometheus, channel="latest/beta")
-    await ops_test.model.deploy(grafana, channel="latest/beta")
-    await ops_test.model.add_relation(prometheus, grafana)
-    await ops_test.model.add_relation(APP_NAME, grafana)
+    await ops_test.model.deploy(prometheus, channel="latest/beta", trust=True)
+    await ops_test.model.deploy(grafana, channel="latest/beta", trust=True)
```

Comment:
I had issues with grafana `latest/beta` as it is very old. I don't see problems in your CI, but it would be good to just upgrade it to `latest/edge` just to avoid future issues. This was suggested by the obs team.